pkg/apiserver/utils/tablecol.go

```diff
@@ -0,0 +1,43 @@
+// Copyright 2021 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package utils
+
+import (
+	"fmt"
+
+	"github.com/jinzhu/gorm"
+
+	"github.com/pingcap/tidb-dashboard/pkg/utils/schema"
+)
+
+var cachedTableColumnsMap map[string][]string = make(map[string][]string)
```

Comment:
We should avoid using the global variable. It would be better to be a `sysschema.CacheService`, which is initialized in apiserver or somewhere else, that allows multiple instances, while passing the current instance to places that need it.